bin/ncp-diag

```diff
@@ -62,38 +62,111 @@ echo "HPB service|$( ncc notify_push:self-test &>/dev/null && echo up || echo do
 echo "Postfix service|$( pgrep -fc postfix &>/dev/null && echo up || echo down )"
 
 # WAN
-echo "internet check|$( ping -W 2 -w 1 -q github.com &>/dev/null && echo ok || echo no )"
+echo "Internet check|$( ping -W 2 -w 1 -q github.com &>/dev/null && echo ok || echo no )"
 
 function is_port_open()
 {
-  local port=$1
-  local public_ip
-  public_ip="$(curl -m4 -4 https://icanhazip.com 2>/dev/null)" || { echo "closed"; return 1; }
-  if [[ "${public_ip}" == "" ]]; then
-    public_ip="$(curl -m4 -6 https://icanhazip.com 2>/dev/null)" || { echo "closed"; return 1; }
-  fi
-  local tmp_file=$(mktemp)
-  local token=$(wget -T2 -t1 -qO- --keep-session-cookies --save-cookies $tmp_file https://portchecker.co | grep -oP "_csrf\" value=\"\K.*\"")
-
-  if [[ "${token}" != "" ]]; then
-    wget -T2 -t1 -qO- --load-cookies $tmp_file https://portchecker.co/check --post-data "target_ip=${public_ip}&port=${port}&_csrf=${token::-1}" \
-    | grep -q '<span class="green">open</span>' && { echo "open"; return 1; }
-  fi
-  rm $tmp_file
-  echo "closed"
+  local port port_url port_check_url ip_url \
```

Comment:
Alright, this is something shellcheck says not to do though, isn't it? 🙏

[SC2155](https://www.shellcheck.net/wiki/SC2155) - Declare and assign separately to avoid masking return values

It's referred to as problematic code?

https://github.com/codacy/codacy-hadolint/blob/master/codacy-hadolint/docs/description/SC2155.md

https://unix.stackexchange.com/a/687856

https://deepsource.io/directory/analyzers/docker/issues/DOK-SC2155